six/modules/c++/cphd/unittests/test_cphd_read_unscaled_int.cpp

```diff
@@ -0,0 +1,307 @@
+/* =========================================================================
+ * This file is part of cphd-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2016, MDA Information Systems LLC
+ *
+ * cphd-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <cstdio>
+#include <stdlib.h>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <memory>
+
+#include <cphd/CPHDReader.h>
+#include <cphd/CPHDWriter.h>
+#include <types/RowCol.h>
+#include <cli/ArgumentParser.h>
+
+#include "TestCase.h"
+
+namespace
+{
+class TempFile
```

Comment:
Let's wait and get the TempFile support from CODA-OSS merged in and then we can wipe this